subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,263 @@
+
+Gradle allows you to skip any task you want based on a condition via the api:org.gradle.api.Task#onlyIf(org.gradle.api.specs.Spec)[] method. The following sample demonstrates how to implement the constraints we just mentioned:
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="publishingMavenConditionally" title="Configuring which artifacts should be published to which repositories">
+    <sourcefile file="build.gradle" snippet="task-config"/>
+    <output args="publish publishToMavenLocal"/>
+</sample>
+++++
+
+You may also want to define your own aggregate tasks to help with your workflow. For example, imagine that you have several publications that should be published to the external repository. It could be very useful to publish all of them in one go without publishing the internal ones.
+
+The following sample demonstrates how you can do this by defining an aggregate task — `publishToExternalRepository` — that depends on all the relevant publish tasks:
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="shorthandTasks" title="Defining your own shorthand tasks for publishing">
+    <sourcefile file="build.gradle" snippet="shorthand-tasks"/>
+</sample>
+++++
+
+This particular sample automatically handles the introduction or removal of the relevant publishing tasks by using api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] with the api:org.gradle.api.publish.maven.tasks.PublishToMavenRepository[] task type. You can do the same with api:org.gradle.api.publish.ivy.tasks.PublishToIvyRepository[] if you're publishing to Ivy-compatible repositories.
+
+[[sec:configuring_publishing_tasks]]
+=== Configuring publishing tasks
+
+The publishing plugins create their non-aggregate tasks after the project has been evaluated, which means you cannot directly reference them from your build script. If you would like to configure any of these tasks, you should use deferred task configuration. This can be done in a number of ways via the project's `tasks` collection.
+
+For example, imagine you want to change where the `generatePomFileFor__PubName__Publication` tasks write their POM files. You can do this by using the api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] method, as demonstrated by this sample:
+
+++++
+<sample dir="userguide/publishing/javaLibrary" id="deferred-configuration-publishing-task" title="Configuring a dynamically named task created by the publishing plugins">
+    <sourcefile file="build.gradle" snippet="configure-generate-task"/>
+</sample>
+++++
+
+The above sample uses a regular expression to extract the name of the publication from the name of the task. This is so that there is no conflict between the file paths of all the POM files that might be generated. If you only have one publication, then you don't have to worry about such conflicts since there will only be one POM file.
+
+=== Terminology
```

Comment:
Interesting choice to put terminology here. I like it.

Reply:
I'm planning on moving all terminology into a single chapter, but didn't want to introduce it with this pull request.